pkg/azuremachine/mutate_update.go

```diff
@@ -0,0 +1,80 @@
+package azuremachine
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"k8s.io/api/admission/v1beta1"
+	"sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+	ctrl "sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/azure-admission-controller/pkg/key"
+	"github.com/giantswarm/azure-admission-controller/pkg/mutator"
+)
+
+type UpdateMutator struct {
+	ctrlClient ctrl.Client
+	logger     micrologger.Logger
+}
+
+type UpdateMutatorConfig struct {
+	CtrlClient ctrl.Client
+	Logger     micrologger.Logger
+}
+
+func NewUpdateMutator(config UpdateMutatorConfig) (*UpdateMutator, error) {
```

Comment:
I think you missed actually creating and wiring an instance of this mutator. Remember that it also needs to be added in the helm chart IIRC

Reply:
Ah yes 😅